doc/pitfalls/data_leakage.rst

```diff
@@ -0,0 +1,157 @@
+.. _data_leakage:
+
+============
+Data leakage
+============
+
+Data leakage occurs when information that would not be available at prediction
+time is used when building the model. This results in overly optimsitic
+performance estimates, for example from :ref:`cross validation
+<cross_validation>`, and thus poorer performance when the model is used
+on actually novel data, for example during production.
+
+A common cause is not keeping the test and train data subsets separate.
+Test data should never be used to make choices about the model.
+The general rule is to never call `fit` on testing data. While this may
+sound obvious, this is easy to miss in some cases, for example when applying
+some pre-processing steps.
+
+Although both train and test data subsets should undergo the same preprocessing
+transformation, it is important that these transformations are only learnt
+from the training data. For example, if you have a
+normalization step where you divide by the average value, the average should
+be the average of the train subset, **not** the average of all the data. If the
+test subset was included in the average calculation, information from the test
+subset is influencing the model. Other types of preprocessing such as
+:ref:`impute` (see :ref:`data_leakage_imputation` below) and
+:ref:`polynomial_features` should also only utilize train data.
```

Comment:
PolynomialFeatures only depends on the shape of the training data, AFAIK, not its values, so it doesn't apply here.

Processing such as decomposition and kernel approximation can be affected. The most problematic transformers for leakage, however, are those that are supervised.